coding_standard_c.txt

```diff
@@ -0,0 +1,147 @@
+
+/*
+ * This block comment could apply to some function and describe its inner
+ * workings.  Notice these sentences have traditional capitalization and
+ * punctuation... that's because it has to be read in a way completely
+ * unlike the Post-It style content of next-line and line comments.
+ */
+
+/*
+ * This line or comment is important... at least more than usual
+ */
+
+/* the next line is regular, but needs a comment */
+
+// This is also acceptable for a line comment.
+
+// This is not an acceptable block comment.
+// These read very imbalanced.
+// Don't do this.
+
+“The size of the comment should be proportional to the size of the code that it refers to. Consequently, properties of code that can fit within a single 24-line screen should not be commented unless they are not obvious. By contrast, even obvious global properties and invariants may need to be made explicit. This doesn't have to be through comments, though. The assert() macro is an excellent executable comment.” [2]  All source files and functions require leading block comments for consistency.
+
+Naming conventions
+Variable names should clearly imply what their values mean... do not rely on the types of variables to do this.  Multiple word names are descriptive and most easily read if words are separated with the underscore character, “_”.  DoNotUseCamelBackStyle because ItIsNotAs easy to read, especially when actual whitespace IsAddedFor syntactical purposes.  It is common practice to keep variable and function names completely lowercase.  Constants are usually lowercase as well, but some groups prefer the standard of all-caps so they have the “eye feel” of older #define macros.
+
+unsigned int table_index = find_index(jacket_table, “color”, COLOR_RED);
+
+Boolean variables should be named to reflect the true condition... even if the variable is used to detect false conditions.
```

Comment:
How do you feel about using explicit boolean types? Should we encourage the use of stdbool instead of ints?